Add support for C2B transaction simulation.

swimlappy opened the pull request:
Hello,

For consideration, I have added support for sending a C2B transaction simulation.  This is something that really helps during testing.  This is a fantastic package and would really like to see if this can be merged in.

Thank you!

swimlappy committed:
Add support for C2B transaction simulation.

SmoDav commented on README.md:
```diff
@@ -129,6 +131,46 @@ $response = \Registrar::register(600000)
+$response = $simulate->request(10)
+    ->from(254722000000)
+    ->usingReference('f4u239fweu')
+    ->setCommand('CustomerPayBillOnline')
```
Add this to a constant. It's better to call `Simulate::CUSTOMER_PAYBILL_ONLINE`

swimlappy commented:
Hello, have you had a chance to review this pull request?  Thanks

SmoDav commented on src/Mpesa/C2B/Simulate.php:
```diff
@@ -0,0 +1,144 @@
+    public function usingReference($reference)
+    {
+        \preg_match('/[^A-Za-z0-9]/', $reference, $matches);
+
```
Also remove the validation as all characters are allowed.

SmoDav commented on src/Mpesa/C2B/Simulate.php:
```diff
@@ -0,0 +1,144 @@
+     */
+    public function setCommand($command)
+    {
+        $valid_commands = ['CustomerPayBillOnline', 'CustomerBuyGoodsOnline'];
```
Add these commands as constants

swimlappy committed:
Update Simulate.php

Updated constants based on review from owner.

swimlappy committed:
Update README.md

Update calls with constants based on review from owner.

swimlappy committed:
Merge pull request #1 from swimlappy/dev-mpesa

Updates requested by owner

swimlappy commented:
Hello, thanks for the review.  I've added in the changes that you have requested and added to this pull request.  Please let me know if these look good.  Thank you

swimlappy committed:
Update README.md

Missed an update to README

swimlappy committed:
Merge pull request #2 from swimlappy/dev-mpesa

Update README.md

swimlappy committed:
Update Simulate.php

Small fix to completely remove validation.

swimlappy committed:
Merge pull request #3 from swimlappy/dev-mpesa

Update Simulate.php

swimlappy committed:
Update README.md

No my best day.  Snuck some extra characters in from a copy/paste.

swimlappy commented:
Hi, do the changes look ok to you?  Thanks

swimlappy commented:
Hello, have you had a chance to review the additional changes?  Thanks
